modules/navier_stokes/src/fvkernels/PINSFVTurbulentAdvection.C

```diff
@@ -0,0 +1,70 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PINSFVTurbulentAdvection.h"
+#include "INSFVEnergyVariable.h"
+#include "PINSFVSuperficialVelocityVariable.h"
+
+registerMooseObject("NavierStokesApp", PINSFVTurbulentAdvection);
+
+InputParameters
+PINSFVTurbulentAdvection::validParams()
+{
+  auto params = INSFVAdvectionKernel::validParams();
+  params.addClassDescription("Advects turbulence energies or dissipation, "
+                             " e.g. rho*TKE. A user may still override what "
+                             "quantity is advected, but the default is rho*turbulent_energy");
+  params.addRequiredParam<MooseFunctorName>(NS::density, "fluid density");
+  return params;
+}
+
+PINSFVTurbulentAdvection::PINSFVTurbulentAdvection(const InputParameters & params)
+  : INSFVAdvectionKernel(params), _rho(getFunctor<ADReal>(NS::density))
+{
+}
+
+ADReal
+PINSFVTurbulentAdvection::computeQpResidual()
```

Comment:
Why do we need a new kernel for turbulent advection? Is there a specific need behind this? Do we need a rho to make sure we are advecting conserved variables? I think we should extend `INSFVScalarFieldAdvection` instead of creating a 80% redundant kernel.